File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        if '0' not in s: return len(s)-1
        if '1' not in s: return len(s)-1
        mx = 0
        for i in range(len(s)-1):
            mx = max(mx, (s[:i+1].count('0')+s[i+1:].count('1')))
        return mx

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
       if (s.find('0') == string::npos) return s.length() - 1;
        if (s.find('1') == string::npos) return s.length() - 1;
        int mx = 0;
        for (size_t i = 0; i < s.length() - 1; ++i) {
            mx = max(mx, static_cast<int>(count(s.begin(), s.begin() + i + 1, '0')) +
                            static_cast<int>(count(s.begin() + i + 1, s.end(), '1')));
        }
        return mx;
    }
};